stdlib/public/runtime/ImageInspection.h

```diff
@@ -30,6 +30,18 @@ namespace swift {
     void *symbolAddress;
   };
 
+#if defined(__ELF__) || defined(__ANDROID__)
+
+struct SectionInfo {
+  uint64_t size;
+  const char *data;
+};
+
+// Called by injected constructors when a dynamic library is loaded.
+void addNewDSOImage(const void *addr);
```

Comment:
Since this is exported from the runtime, we should give it an `extern "C"` name like `swift_addNewDSOImage`, so that the runtime ABI isn't dependent on the C++ ABI.